x509/x509.go

```diff
@@ -811,7 +816,12 @@ func (p *PEMEncodedCertificateAndKey) MarshalYAML() (interface{}, error) {
 	}
 
 	aux.Crt = base64.StdEncoding.EncodeToString(p.Crt)
-	aux.Key = base64.StdEncoding.EncodeToString(p.Key)
+
+	if string(p.Key) == Redacted {
+		aux.Key = Redacted
+	} else {
+		aux.Key = base64.StdEncoding.EncodeToString(p.Key)
```

Comment:
I think we need to have symmetric handling in `UnmarshalYAML`

Reply:
true, done